cfme/tests/infrastructure/test_timelines.py

```diff
@@ -103,99 +152,215 @@ def _power_on(self):
-                                                                    found_events])))
+            try:
+                if not policy_events:
+                    # Special case for create event
+                    if hasattr(evt, 'destination_vm') and evt.destination_vm in self.vm.name:
+                        found_events.append(evt)
+                        break
+                    # Other events
+                    elif evt.source_vm in self.vm.name and evt.event_type in self.tl_event:
+                        found_events.append(evt)
+                        break
+                else:
+                    if evt.event_type in self.tl_event and evt.target in self.vm.name:
+                        found_events.append(evt)
+                        break
+            except AttributeError as err:
+                logger.warn('Issue with TimelinesEvent: %r .Faulty event: %r', str(err), str(evt))
+                continue
+
+        logger.info('found events on %r :\n %s', target, '\n'.join([repr(e) for e in found_events]))
+
         return len(found_events)
 
-    def catch_in_timelines(self, soft_assert, targets):
+    def catch_in_timelines(self, soft_assert, targets, policy_events=False):
         if targets:
             for target in targets:
                 try:
-                    wait_for(self._check_timelines, [target], timeout='5m', fail_condition=0)
+                    wait_for(self._check_timelines,
+                             [target, policy_events],
+                             timeout='7m',
+                             fail_condition=0)
                 except TimedOutError:
                     soft_assert(False, '0 occurrence of {} found on the timeline of {}'.format(
                         self.event, target))
         else:
             raise ValueError('Targets must not be empty')
 
 
-@pytest.mark.rhv2
-def test_timeline_events(new_vm, soft_assert):
-    events_list = ['create', 'stop', 'start', 'suspend', 'start']
+def test_infra_timeline_create_event(new_vm, soft_assert):
+    event = 'create'
+    vm_event = VMEvent(new_vm, event)
     targets = (new_vm, new_vm.cluster, new_vm.host, new_vm.provider)
-    for event in events_list:
-        vm_event = VMEvent(new_vm, event)
-        logger.info('Will generate event %r on machine %r', event, new_vm.name)
-        wait_for(vm_event.emit, timeout='5m', message='Event {} did timeout'.format(event))
-        vm_event.catch_in_timelines(soft_assert, targets)
+    logger.info('Will generate event %r on machine %r', event, new_vm.name)
+    wait_for(vm_event.emit, timeout='7m', message='Event {evt}failed'.format(evt=event))
+    vm_event.catch_in_timelines(soft_assert, targets)
 
 
-def test_infra_diagnostic_timeline_events(new_vm, soft_assert, mark_vm_as_appliance):
-    events_list = ['create']
+@pytest.mark.meta(blockers=[BZ(1542962, forced_streams=['5.8'])])
+def test_infra_timeline_policy_event(new_vm, control_policy, soft_assert):
```

Comment:
please add doc string with short explanation what is tested by test.

Reply:
ok